Code/Tools/AssetProcessor/native/tests/assetmanager/AssetProcessorManagerTest.cpp

```diff
@@ -2150,6 +2150,22 @@ TEST_F(PathDependencyTest, WildcardProductPathDependenciesWithBackSlash_Deferred
         /*buildDependenciesFirst*/ false);
 }
 
+TEST_F(PathDependencyTest, Wildcard_ResolvingTwice_DependenciesNotDuplicated)
+{
+    // Regression test: make sure resolving the dependencies twice doesn't result in duplicate entries in the database
+
+    RunWildcardTest(
+        /*useCorrectDatabaseSeparator*/ true,
+        AssetBuilderSDK::ProductPathDependencyType::ProductFile,
+        /*buildDependenciesFirst*/ true
+    );
+    RunWildcardTest(
+        /*useCorrectDatabaseSeparator*/ true,
+        AssetBuilderSDK::ProductPathDependencyType::ProductFile,
+        /*buildDependenciesFirst*/ false);
```

Comment:
I can see the query is run twice here, but how does it detect that the duplicate entries?

Reply:
RunWildcardTest does a check at the end to verify the entries in the DB match what its expecting.  When running it twice without the fix, it ended up with more entries than expected so it failed.